ccl/oidcccl: TestOIDCEnabled failed

ccl/oidcccl.TestOIDCEnabled [failed](https://mesolite.cluster.engflow.com/invocations/default/b60663b9-8cc1-4ceb-b5fd-7eec2c524c36?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL29pZGNjY2w6b2lkY2NjbF90ZXN0) on release-25.1.9-rc @ [da9b0ac6f6220c719b417a4e15d975773a8e74be](https://github.com/cockroachdb/cockroach/commits/da9b0ac6f6220c719b417a4e15d975773a8e74be):


```
=== RUN   TestOIDCEnabled
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestOIDCEnabled920431325
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    authentication_oidc_test.go:202: could not issue GET request to callback endpoint: Get "https://127.0.0.1:38169/oidc/v1/callback?state=GiBrtESVugwYowFQqrJYQnrZwU-IVCDbRUQtvRM5d-3I6iIgdL7yBNIzc3GBZmrKFn-3rs2P22yCwjJNsn6gBbbybZA%3D": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestOIDCEnabled920431325
--- FAIL: TestOIDCEnabled (83.05s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/product-security
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestOIDCEnabled.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-52585

--- a/pkg/ccl/oidcccl/authentication_oidc_test.go
+++ b/pkg/ccl/oidcccl/authentication_oidc_test.go
@@ -145,6 +145,9 @@ func TestOIDCEnabled(t *testing.T) {
 	client, err := testCertsContext.GetHTTPClient()
 	require.NoError(t, err)
 
+	// Set a reasonable timeout for the client to prevent flakiness under stress.
+	client.Timeout = 30 * time.Second
+
 	// Don't follow redirects so we can inspect the 302
 	client.CheckRedirect = func(req *http.Request, via []*http.Request) error {
 		return http.ErrUseLastResponse

